docs/design/persistent-storage.md

```diff
@@ -0,0 +1,222 @@
+# Persistent Stores
+
+This document proposes a model for managing persistent, cluster-scoped storage for applications requiring long lived data.
+
+### tl;dr
+
+`PersistentStore` objects contain a VolumeSource with a lifecycle greater than the pods it is used by.  Many types of VolumeSources can be implemented as volume plugins, with each implementing an interface for use within a controller.  A pod with persistence requirements (specifically pod.spec.volumes.volume.persistenceRequirements) is bound to a matching VolumeSource from a pool of available stores.  PersistentStore objects give that VolumeSource longevity.
+
+A `PersistentStoreController` manages the creation, attachment, and reclamation of PersistentStore objects for one VolumeSource.  Implementations must be pluggable to support many types of controllers for many types of VolumeSources. The controller can advise the scheduler on host selection, keeping volume rules contained within the provider.  All cloud provider credentials are isolated to the store controller's host.
+
+`PersistentStoreBinder` watches for pods with persistence requirements and matches the request with what is available in the pool.
+
+`PersistenceRequirements` expresses the capabilities and performance attributes of the underlying VolumeSource.  Allow discovery
+of storage by its capabilities and use those capabilities as pod storage requirements.
+
+Kubernetes makes no guarantees at runtime that the underlying storage exists or is available.  High availability is left to the storage provider.
+
+## Goals
+
+* Allow administrators to describe available storage
+* Allow pod authors to discover and request persistence for their volumes
+* Enforce security through access control lists and securing storage to the same namespace as the pod volume
+* Enforce quotas through admission control
+* Ensure developers can rely on storage being available without being closely bound to a particular disk, server, network, or storage device.
+
+
+#### Allow admins to describe and create available storage
+
+Admins can describe storage using a `PersistentStoreController` for each type of VolumeSource being managed or create storage directly by using the `PersistentStore` API.  Storage created manually still requires a PersistentStoreController and accompanying interface to allow management by the cluster.
+
+
+```
+	
+	PersistentStore
+		TypeMeta
+		ObjectMeta
+			Name - internal name
+			Labels - labeled like Volume.PersistenceRequirements.Selector
+		Spec
+			VolumeSource -- backed by one of these:
+				HostPath
```

Comment:
My question was whether you're proposing that local storage be managed in this same way, and if so that the act of binding to a PS with local storage would influence scheduling?  That's a somewhat different proposal than network attached storage.  Or maybe not, given things like max PDs per VM... ?